panic when list image that be pulled by digest.

panic: runtime error: index out of range

I can fix it.

Log is below.
```
root@ubuntu:/home/cloud/go/src/github.com/kubernetes-incubator/cri-containerd# crictl pull docker.io/library/busybox@sha256:be3c11fdba7cfe299214e46edc642e09514dbb9bbefcd0d3836c05a1e0cd0642
DEBU[0000] PullImageRequest: &PullImageRequest{Image:&ImageSpec{Image:docker.io/library/busybox@sha256:be3c11fdba7cfe299214e46edc642e09514dbb9bbefcd0d3836c05a1e0cd0642,},Auth:nil,SandboxConfig:nil,} 
DEBU[0002] PullImageResponse: &PullImageResponse{ImageRef:sha256:c30178c5239f2937c21c261b0365efcda25be4921ccb95acd63beeeb78786f27,} 
Image is update to date for sha256:c30178c5239f2937c21c261b0365efcda25be4921ccb95acd63beeeb78786f27
root@ubuntu:/home/cloud/go/src/github.com/kubernetes-incubator/cri-containerd# crictl images
DEBU[0000] ListImagesRequest: &ListImagesRequest{Filter:&ImageFilter{Image:&ImageSpec{Image:,},},} 
DEBU[0000] ListImagesResponse: &ListImagesResponse{Images:[&Image{Id:sha256:c30178c5239f2937c21c261b0365efcda25be4921ccb95acd63beeeb78786f27,RepoTags:[],RepoDigests:[docker.io/library/busybox@sha256:be3c11fdba7cfe299214e46edc642e09514dbb9bbefcd0d3836c05a1e0cd0642],Size_:701277,Uid:nil,Username:,}],} 
panic: runtime error: index out of range

goroutine 1 [running]:
main.glob..func10(0xc4200e4780, 0x0, 0xc4200e4780)
	/home/cloud/go/src/github.com/kubernetes-incubator/cri-tools/cmd/crictl/image.go:135 +0x5b1
github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli.HandleAction(0x1375cc0, 0x15c3870, 0xc4200e4780, 0xc42006f400, 0x0)
	/home/cloud/go/src/github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli/app.go:485 +0xd4
github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli.Command.Run(0x155c234, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x156176e, 0xb, 0x0, ...)
	/home/cloud/go/src/github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli/command.go:207 +0xb72
github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli.(*App).Run(0xc420087ba0, 0xc42000c1e0, 0x2, 0x2, 0x0, 0x0)
	/home/cloud/go/src/github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli/app.go:250 +0x7d0
main.main()
	/home/cloud/go/src/github.com/kubernetes-incubator/cri-tools/cmd/crictl/main.go:195 +0xaaf
```

--- a/cmd/crictl/image.go
+++ b/cmd/crictl/image.go
@@ -125,14 +125,12 @@ var listImageCommand = cli.Command{
 					printHeader = false
 					fmt.Fprintln(w, "IMAGE\tTAG\tIMAGE ID\tSIZE")
 				}
-				repoTags := "<none>"
-				if image.RepoTags != nil {
-					repoTags = image.RepoTags[0]
-				}
-				repoTagsPair := strings.Split(repoTags, ":")
+				repoTagPairs := normalizeRepoTagPair(image.RepoTags, image.RepoDigests)
 				size := units.HumanSizeWithPrecision(float64(image.GetSize_()), 3)
 				trunctedImage := strings.TrimPrefix(image.Id, "sha256:")[:truncatedImageIDLen]
-				fmt.Fprintf(w, "%s\t%s\t%s\t%s\n", repoTagsPair[0], repoTagsPair[1], trunctedImage, size)
+				for _, repoTagPair := range repoTagPairs {
+					fmt.Fprintf(w, "%s\t%s\t%s\t%s\n", repoTagPair[0], repoTagPair[1], trunctedImage, size)
+				}
 				continue
 			}
 			fmt.Printf("ID: %s\n", image.Id)
@@ -268,6 +266,29 @@ func getAuth(creds string) (*pb.AuthConfig, error) {
 		Username: username,
 		Password: password,
 	}, nil
+}
+
+// Ideally repo tag should always be image:tag.
+// The repoTags is nil when pulling image by repoDigest,Then we will show image name instead.
+func normalizeRepoTagPair(repoTags []string, repoDigests []string) (repoTagPairs [][]string) {
+	if len(repoTags) == 0 {
+		if len(repoDigests) == 0 {
+			repoTagPairs = append(repoTagPairs, []string{"errorRepoDigest", "errorRepoDigest"})
+			return
+		}
+		imageName := strings.Split(repoDigests[0], "@")[0]
+		repoTagPairs = append(repoTagPairs, []string{imageName, "<none>"})
+		return
+	}
+
+	for _, repoTag := range repoTags {
+		if idx := strings.Index(repoTag, ":"); idx == -1 {
+			repoTagPairs = append(repoTagPairs, []string{"errorRepoTag", "errorRepoTag"})
+			continue
+		}
+		repoTagPairs = append(repoTagPairs, strings.Split(repoTag, ":"))
+	}
+	return
 }
 
 // PullImage sends a PullImageRequest to the server, and parses

